src/basic/path-util.h

```diff
@@ -49,6 +49,7 @@ int path_compare(const char *a, const char *b) _pure_;
 bool path_equal(const char *a, const char *b) _pure_;
 bool path_equal_or_files_same(const char *a, const char *b, int flags);
 char* path_join(const char *root, const char *path, const char *rest);
+char* path_join_many(const char *first, ...);
```

Comment:
I'd prefer to have this wrapped like `strjoina` and others (with the real function renamed and `#define path_join_many(x, ...) _path_join_many(x, __varargs__, NULL)`.